modules/cudf/test/series/reduce/cummax-tests.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import '../../jest-extensions';
+
+import {setDefaultAllocator} from '@nvidia/cuda';
+import {
+  Bool8,
+  Float32,
+  Float64,
+  Int16,
+  Int32,
+  Int64,
+  Int8,
+  Series,
+  Uint16,
+  Uint32,
+  Uint64,
+  Uint8
+} from '@rapidsai/cudf';
+import {DeviceBuffer} from '@rapidsai/rmm';
+
+setDefaultAllocator((byteLength: number) => new DeviceBuffer(byteLength));
+
+const numbers        = [4, 2, null, 5, 1, 1];
+const float_with_NaN = [4, 2, NaN, 5, 1, 1];
+const bools          = [true, false, null, false, true];
+const bigints        = [4n, 2n, null, 5n, 1n, 1n];
+
+function testNumberAny<T extends Int8|Int16|Int32|Uint8|Uint16|Uint32|Float32|Float64>(
+  type: T, data: (T['scalarType']|null)[], skipna = true) {
+  const expected = skipna ? [4, 4, null, 5, 5, 5]  //
+                          : [4, 4, null, null, null, null];
```

Comment:
Question here (and other tests) should the `float-nan` cases match this output exactly as well? Making that so requires calling `nansToNulls` on float series unconditionally. Python cudf on calls `nans_to_nulls` when `skipna=True` (but also seems to treat nans as nulls from the start)

Reply:
If I'm understanding correctly, calling `nansToNulls` unconditionally seems a bit odd to me. Be curious to hear what others think.

Reply:
well, in a nutshell, this difference is why:

### Python cudf
```
In [3]: a = cudf.Series([4, 2, np.NaN, 5, 1, 1.0])

In [4]: a
Out[4]: 
0     4.0
1     2.0
2    <NA>
3     5.0
4     1.0
5     1.0
dtype: float64

In [5]: a.has_nulls
Out[5]: True
```

#### node-rapids cudf
```
> a = Series.new([2, 1, NaN, 3, 4.0])
Float64Series { _col: Column {} }
> a.hasNulls
false
```

---

To be honest, given the above I am not sure why Python calls `nans_to_nulls` at all. Maybe it is just cruft let over from the past?

Reply:
Maybe this question is better to raise in a dedicated issue?